server/session/handler_mob_equipment.go

```diff
@@ -21,7 +21,17 @@ func (*MobEquipmentHandler) Handle(p packet.Packet, s *Session) error {
 		// This window ID is expected, but we don't handle it.
 		return nil
 	case protocol.WindowIDInventory:
-		return s.UpdateHeldSlot(int(pk.InventorySlot), stackToItem(pk.NewItem.Stack))
+		// The item the client claims to have must be identical to the one we have registered server-side.
+		actual, _ := s.inv.Item(int(pk.InventorySlot))
+		clientSide := stackToItem(pk.NewItem.Stack)
+		if !actual.Equal(clientSide) {
+			// Only ever debug these as they are frequent and expected to happen whenever client and server get
+			// out of sync.
+			s.log.Debugf("failed processing packet from %v (%v): *packet.MobEquipment: client-side item must be identical to server-side item, but got differences: client: %v vs server: %v", s.conn.RemoteAddr(), s.c.Name(), clientSide, actual)
```

Comment:
Technically this didn’t fail though as it continues to process the packet

Reply:
this is just a potentially useful debug imo